rct229/rulesets/ashrae9012019/section10/section10rule7.py

```diff
@@ -0,0 +1,259 @@
+                for zone_id in hvac_data["zone_list"]
+            ]
+            for hvac_id, hvac_data in get_hvac_zone_list_w_area_dict(rmd_b).items()
+        }
+
+        return {
+            "baseline_system_types_dict": {
+                system_type: [
+                    system_id
+                    for system_id in system_list
+                    if system_id not in baseline_sys_5_6_serve_more_than_one_flr_list
+                ]
+                for system_type, system_list in baseline_system_types_dict.items()
+                if system_type in APPLICABLE_SYS_TYPES and system_list
+            },
+            "baseline_system_zones_served_dict": baseline_system_zones_served_dict,
+        }
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section10Rule7.HVACRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=False, BASELINE_0=True, PROPOSED=False
+                ),
+                required_fields={
+                    "$": ["cooling_system"],
+                    "cooling_system": [
+                        "efficiency_metric_types",
+                        "efficiency_metric_values",
+                    ],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            hvac_b = context.BASELINE_0
+            hvac_id_b = hvac_b["id"]
+            baseline_system_types_dict = data["baseline_system_types_dict"]
+
+            return any(
+                hvac_id_b in baseline_system_types_dict[system_type]
+                for system_type in baseline_system_types_dict
+            )
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.BASELINE_0
+            cooling_system_b = hvac_b["cooling_system"]
+            baseline_system_types_dict_b = data["baseline_system_types_dict"]
+            hvac_zone_list_w_area_dict_b = data["baseline_system_zones_served_dict"]
+            zone_list_b = hvac_zone_list_w_area_dict_b[hvac_b.id]
+            is_zone_agg_factor_undefined_and_needed = False
+
+            hvac_system_type_b = next(
+                (
+                    system_type
+                    for system_type, hvac_b_ids in baseline_system_types_dict_b.items()
+                    if hvac_b["id"] in hvac_b_ids
+                ),
+                None,
+            )
+
+            total_cool_capacity_b = cooling_system_b.get("rated_total_cool_capacity")
+            if total_cool_capacity_b is None:
+                total_cool_capacity_b = cooling_system_b.get(
+                    "design_total_cool_capacity"
+                )
+            if total_cool_capacity_b is not None and hvac_system_type_b in [
+                HVAC_SYS.SYS_3,
+                HVAC_SYS.SYS_3B,
+                HVAC_SYS.SYS_4,
+            ]:
+                hvac_zone_aggregation_factor = zone_list_b[0].get("aggregation_factor")
```

Comment:
`aggregation_factor` is something the RPD must provide in certain situations. The reason I carried it into `manual_check_required` is so that we can provide a unique message if the aggregation_factor is needed, it is undefined, and the most efficient value from the data table was modeled. This is our way of telling a reviewer that the most stringent value was used and it is likely acceptable, but we could not provide a PASS outcome.

Reply:
Yes the exact scenario is if it is baseline system 3 or 4 AND the `aggregation_factor` is missing AND `modeled_efficiency == most_conservative_efficiency` then a custom undetermined message is used. If it is not system 3 or 4 then `aggregation_factor` does not matter for this rule. If modeled_efficiency != most_conservative_efficiency then a different undetermined message is used. This is all in the mod version of the RDS

Reply:
Ok, got it. Then I think the `.get` is appropriate in this code. Thanks for the clarification.